mobile/android/Interface.cpp

```diff
@@ -51,22 +110,83 @@ Java_org_iota_mobile_Interface_iota_1sign_1address_1gen(
 
 /*
  * Class:     org_iota_mobile_Interface
- * Method:    generateSignature
+ * Method:    iota_sign_address_gen_trits
+ * Signature: ([Ljava/lang/Byte;II)[Ljava/lang/Byte;
+ */
+JNIEXPORT jbyteArray JNICALL
+Java_org_iota_mobile_Interface_iota_1sign_1address_1gen_1trits(
+    JNIEnv* env, jclass thiz, jbyteArray jseed, jint index, jint security) {
+  const trit_t* seed = (trit_t*)env->GetByteArrayElements(jseed, 0);
+
+  trit_t* address = iota_sign_address_gen_trits(seed, index, security);
+  std::memset((void*)seed, 0, 243);
+
+  jbyteArray out = env->NewByteArray(243);
+  env->SetByteArrayRegion(out, 0, 243, (const jbyte*)address);
+  free(address);
+
+  return out;
+}
+
+/*
+ * Class:     org_iota_mobile_Interface
+ * Method:    iota_sign_signature_gen_trytes
  * Signature: (Ljava/lang/String;IILjava/lang/String;)Ljava/lang/String;
  */
 JNIEXPORT jstring JNICALL
-Java_org_iota_mobile_Interface_iota_1sign_1signature_1gen(
+Java_org_iota_mobile_Interface_iota_1sign_1signature_1gen_trytes(
```

Comment:
I think this needs to be changed to:
```suggestion
Java_org_iota_mobile_Interface_iota_1sign_1signature_1gen_1trytes(
```
to avoid this error: 

```
02-12 18:54:41.702  7712  7712 E AndroidRuntime: java.lang.UnsatisfiedLinkError: No implementation found for java.lang.String org.iota.mobile.Interface.iota_sign_signature_gen_trytes(java.lang.String, int, int, java.lang.String) (tried Java_org_iota_mobile_Interface_iota_1sign_1signature_1gen_1trytes and Java_org_iota_mobile_Interface_iota_1sign_1signature_1gen_1trytes__Ljava_lang_String_2IILjava_lang_String_2)
02-12 18:54:41.702  7712  7712 E AndroidRuntime: 	at org.iota.mobile.Interface.iota_sign_signature_gen_trytes(Native Method)
```